modules/http-client-rest/src/main/java/com/devonfw/module/httpclient/common/impl/rest/SyncServiceClientStubImpl.java

```diff
@@ -0,0 +1,306 @@
+package com.devonfw.module.httpclient.common.impl.rest;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublisher;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpRequest.Builder;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.nio.file.Path;
+import java.time.temporal.Temporal;
+import java.util.function.Consumer;
+
+import javax.ws.rs.core.MediaType;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.devonfw.module.httpclient.common.impl.ServiceHttpClient;
+import com.devonfw.module.service.common.api.client.async.ServiceClientInvocation;
+import com.devonfw.module.service.common.api.client.async.ServiceClientStub;
+import com.devonfw.module.service.common.api.client.context.ServiceContext;
+import com.devonfw.module.service.common.base.client.ServiceClientPerformanceLogger;
+import com.devonfw.module.service.common.base.client.SyncServiceClientInvocationImpl;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Implementation of {@link ServiceClientStub} and {@link InvocationHandler} that acts as dynamic proxy for service API
+ * to capture {@link ServiceClientInvocation}s.
+ *
+ * @param <S> type of the {@link ServiceContext#getApi() service API}.
+ * @since 2021.08.003
+ */
+public class SyncServiceClientStubImpl<S> implements ServiceClientStub<S>, InvocationHandler {
+  private static final Logger LOG = LoggerFactory.getLogger(SyncServiceClientStubImpl.class);
+
+  private final ServiceContext<S> context;
+
+  private S proxy;
+
+  /** {@link ServiceHttpClient} to use. */
+  protected final ServiceHttpClient client;
+
+  private final ObjectMapper objectMapper;
+
+  private final RestServiceMetadata<S> serviceMetadata;
+
+  /** The {@link #setErrorHandler(Consumer)} */
+  private Consumer<Throwable> errorHandler;
```

Comment:
should also be removed as this only  makes sense for async usage. There is no API to set this in sync usage and the sync API is designed to throw errors immediately when received.